src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -311,357 +336,364 @@ void func_80BA8DF4(EnBaba* this, PlayState* play) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, this->animIndex);
+            this->flags |= ENBABA_AUTOTALK;
+            this->actionFunc = EnBaba_Idle;
         } else {
             Actor_MarkForDeath(&this->actor);
             return;
         }
     } else {
-        this->unk_40A |= 2;
-        if (ENBABA_GET_C000(&this->actor) == ENBABA_C000_2) {
+        this->flags |= ENBABA_VISIBLE;
+        if (ENBABA_GET_TYPE(&this->actor) == ENBABA_TYPE_SWAY) {
             this->actor.flags &= ~ACTOR_FLAG_1;
-            this->unk_40C = 5;
-            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 5);
-            this->actionFunc = func_80BA9E00;
-        } else if ((ENBABA_GET_3F00(&this->actor)) != ENBABA_3F00_3F) {
-            this->unk_40C = 2;
-            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 2);
-            this->actionFunc = func_80BA9E10;
+            this->animIndex = BABA_ANIM_SWAY;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, this->animIndex);
+            this->actionFunc = EnBaba_DoNothing;
+        } else if (ENBABA_GET_PATH_INDEX(&this->actor) != 0x3F) {
+            this->animIndex = BABA_ANIM_WALKING_HOLDING_BAG;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, this->animIndex);
+            this->actionFunc = EnBaba_Walk;
         } else {
-            this->unk_40C = 0;
-            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 0);
-            this->actionFunc = func_80BA9E48;
+            this->animIndex = BABA_ANIM_IDLE_HOLDING_BAG;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, this->animIndex);
+            this->actionFunc = EnBaba_FaceForward;
         }
     }
 }
 
-void func_80BA9758(EnBaba* this, PlayState* play) {
-    if ((this->unk_40A & 8) || (this->unk_144 != NULL) || func_80BA8820(this, play)) {
+void EnBaba_Idle(EnBaba* this, PlayState* play) {
+    if ((this->flags & ENBABA_AUTOTALK) || (this->bombShopkeeper != NULL) || EnBaba_FindBombShopkeeper(this, play)) {
         if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
-            func_80BA886C(this, play);
-            if (this->unk_40A & 8) {
+            EnBaba_HandleConversation(this, play);
+            if (this->flags & ENBABA_AUTOTALK) {
                 this->actor.flags &= ~ACTOR_FLAG_10000;
             }
-            this->actionFunc = func_80BA98EC;
+            this->actionFunc = EnBaba_Converse;
         } else if (this->actor.xzDistToPlayer < 100.0f) {
-            if (this->unk_40A & 8) {
+            if (this->flags & ENBABA_AUTOTALK) {
                 this->actor.flags |= ACTOR_FLAG_10000;
             }
             func_800B8614(&this->actor, play, 100.0f);
         }
     }
 }
 
-void func_80BA9848(EnBaba* this, PlayState* play) {
-    u8 temp_v0 = Message_GetState(&play->msgCtx);
+void EnBaba_FollowScheduleConverse(EnBaba* this, PlayState* play) {
```

Comment:
So this is different and actually related to the third question in your main comment. The thing is that `EnBaba` has a particular type, given by the params, where the actor just follows the schedule given by the script. In particular I named this the `ENBABA_TYPE_FOLLOW_SCHEDULE` in the enum. This particular function is used only by that type, so the `FollowSchedule` has more to do with the type of the actor, rather than the actual schedule itself. I'll agree that this is a bit confusing, but I couldn't think of a better name for the `type`. If you got any suggestions feel free to let me know.

Reply:
Hmm, I quite like `FollowSchedule` for the actual action function name, as that is the action the actor is doing, following their schedule. If at all possible I'd rather find a way to help distinguish `HandleSchedule` which isn't an action function, but in this case holds most of the logic of how to follow the schedule. I'm also not sure if this will be a more used pattern. For example Bjt doesn't seem to use this pattern, but just has all the logic in the `Wait/Idle` function.

Reply:
Yeah, although I suspect that ??? is atypical here since it's so simple. Could look at something like the Postman, or Anju's Mother.

Reply:
So it does seem to be a more general pattern. Looks like Sakon, the Postman, Dampei, and the milk bar bartender follow this pattern.

Reply:
I would say it does. What you call `EnNb_Wait`, I would call `EnNb_FollowSchedule` (it is the action function that gets set when the actor wants to follow the schedule and is what calls RunScript), than `func_80BC0D1C` would be what I would call `EnNb_HandleSchedule` (it holds the logic of actually following the schedule based on the processed output)

Reply:
Oh I forgot to mention (also don't know if you already saw it above), but I originally had `HandleSchedule` as `HandleScheduleOutput` https://github.com/zeldaret/mm/pull/825#discussion_r912521654 . Though I agree with engineer that it doesn't really need the Output, and I think shortening it to `HandleSchedule` was best.